src/main/java/calculator/StringAddCalculator.java

```diff
@@ -0,0 +1,66 @@
+package calculator;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    public static int splitAndSum(String text) {
+        if (isBlank(text))
+            return 0;
+
+        Integer values = customSplitAndSum(text);
+        if (values != null)
+            return values;
+
+        return sum(split(text));
+    }
+
+    private static boolean isBlank(String text) {
+        return text == null || text.isEmpty();
+    }
+
+    private static Integer customSplitAndSum(String text) {
+        Matcher m = Pattern.compile("//(.)\n(.*)").matcher(text);
+
+        if (!m.find())
+            return null;
+
+        String customDelimiter = m.group(1);
+        String[] values = m.group(2).split(customDelimiter);
+
+        return sum(values);
+    }
+
+    private static String[] split(String text) {
+        return text.split("[,:]");
+    }
+
+    private static int sum(String[] values) {
+        int sum = 0;
+
+        for (int number : toInts(values)) {
+            sum += number;
+        }
+
+        return sum;
+    }
+
+    private static int[] toInts(String[] values) {
```

Comment:
```suggestion
    private static List<Integer> toInts(List<String> values) {
```

[이펙티브 자바 아이템 28](https://madplay.github.io/post/prefer-lists-to-arrays) 에서는 배열보다는 리스트를 사용하기를 권장합니다.
이번 기회에 이펙티브 아이템 28에 대해서 공부해보시고,
언제 배열을 쓰고 언제 리스트를 쓰면 좋을지 본인만의 기준을 코멘트로 남겨주시면 좋을 것 같습니다!

Reply:
명확하게 고정된 데이터일 경우 시간, 공간 복잡도 면에서 유리한 배열을 쓰는 편입니다.

Reply:
리턴타입만 List형으로 변경하였습니다. 본 실습에서 파라미터 List<String>은 split을 다시 한 번 List로 변환하는 과정을 거치는 것이 비효율적이라고 판단되어 배열을 그대로 두었습니다.